src/routes/index.js

```diff
@@ -1,28 +1,244 @@
+          status: statusCodes.UNPROCESSABLE_ENTITY,
+          pointer: 'username',
+        })
+      } else if (!password || password === '') {
+        throw new ErrorResponse({
+          title: 'Password is required.',
+          code: codes.ERR_PARAMETER_REQUIRED,
+          status: statusCodes.UNPROCESSABLE_ENTITY,
+          pointer: 'password',
+        })
+      }
+
+      // TODO: Generate user instead with a random username and password
+      const user = (await User.query(username, { index: 'usersIndex' }))
+        .Items[0]
+
+      if (!user) {
+        throw new ErrorResponse({
+          title: "Username doesn't exist.",
+          code: codes.ERR_NOT_FOUND, //
+          status: statusCodes.NOT_FOUND,
+          pointer: 'username',
+        })
+      }
+
+      const passwordsMatch = Password.compare(user.password, password)
+
+      if (!passwordsMatch) {
+        throw new ErrorResponse({
+          title: 'Username or password is incorrect.',
+          code: codes.ERR_CONFLICT,
+          status: statusCodes.CONFLICT,
+          pointer: 'password',
+        })
+      }
+
+      delete user.password
+
+      reply.send({
+        token: signToken(user.id),
+        data: user,
+      })
+    })
+  )
 
-const routes = {
-  createChat: {
-    method: 'POST',
-    url: '/chat',
-    schema,
-    handler: async (request, reply) => {
-      try {
-        reply.send({ message: 'Chat(WIP)', data })
-      } catch (error) {
-        console.log('error =>', error)
+  // Create chats
+  fastify.post(
+    '/chat',
+    asyncHandler(async (request, reply) => {
+      const { chatName } = request.body
+
+      if (!chatName || chatName === '') {
+        throw new ErrorResponse({
+          title: 'Please provide a chatName.',
+          code: codes.ERR_PARAMETER_REQUIRED,
+          status: statusCodes.UNPROCESSABLE_ENTITY,
+          pointer: 'chatName',
+        })
       }
-    },
-  },
+
+      const chatId = uuidv4()
+
+      // TODO: Add unique constraint for chatName
+      await ChatMessage.put({
+        chatId,
+        chatName,
+      })
+
+      reply.send({
+        data: {
+          chatId,
+          chatName,
+        },
+      })
+    })
+  )
+
+  // Get Messages
+  fastify.get(
+    '/chat/:id/messages',
+    asyncHandler(async (request, reply) => {
+      const chatId = request.params.id
+
+      if (!chatId)
+        connection.socket.send(
+          JSON.stringify(
+            new ErrorResponse({
+              title: 'chatId is required',
+              status: statusCodes.BAD_REQUEST,
+              code: codes.ERR_PARAMETER_REQUIRED,
+              parameter: 'chatId',
+            })
+          )
+        )
+
+      // TODO: Implement pagination
+      const messages = (await ChatMessage.query(chatId)).Items
+
+      //  TODO: Define response format
+      reply.send({
+        data: messages,
+      })
+    })
+  )
+
+  // Send Message
+  fastify.get(
+    '/chat/:id/send',
+    { websocket: true },
```

Comment:
Does `{ websocket: true }` indicate that this is a websocket-only endpoint?

Reply:
Why does the previous route not include this option?

Reply:
{ websocket: true }, and this is the only websocket-only endpoint